storage/src/rocksdb_client.cpp

```diff
@@ -646,6 +647,121 @@ void Client::removeAllCheckpoints() const {
+      all_mem_tables_ram_usage_(metrics_.RegisterAtomicGauge("storage_rocksdb_mem_tables_ram_usage", 0)),
+      all_column_families_block_cache_ram_usage_(
+          metrics_.RegisterAtomicGauge("storage_rocksdb_column_families_block_cache_ram_usage", 0)),
+      indexes_and_filters_ram_usage_(metrics_.RegisterAtomicGauge("storage_rocksdb_indexes_and_filter_ram_usage", 0)),
+      rocksdb_total_ram_usage_(metrics_.RegisterAtomicGauge("storage_rocksdb_total_ram_usage", 0)) {
+  for (const auto &pair : ::rocksdb::TickersNameMap) {
+    if (std::find(tickers.begin(), tickers.end(), pair.first) != tickers.end()) {
+      auto metric_suffix = pair.second;
+      std::replace(metric_suffix.begin(), metric_suffix.end(), '.', '_');
+      active_tickers_.emplace(pair.first, metrics_.RegisterAtomicGauge("storage_" + metric_suffix, 0));
+    }
+  }
+  metrics_.Register();
+}
+
+void Client::RocksDbStorageMetrics::setMetricsDataSources(std::shared_ptr<::rocksdb::SstFileManager> sourceSstFm,
+                                                          std::shared_ptr<::rocksdb::Statistics> sourceStatistics) {
+  sstFm = sourceSstFm;
+  statistics = sourceStatistics;
+
+  update_metrics_ =
+      std::make_unique<concord::util::PeriodicCall>([this]() { updateMetrics(); }, update_metrics_interval_msec_);
+}
+
+// periodically running function to update metrics. this func and its sub-funcs aren't
+// thread safe. It's the Client responsibility to temporarily disable this periodic func
+// if configs change during runtime.
+void Client::RocksDbStorageMetrics::updateMetrics() {
+  static size_t entry_count{0};
+
+  // if isn't initialized yet or disabled - return
+  if (!sstFm || !statistics) return;
+
+  // we don't update mem usage every call since it's pretty heavy on resources and isn't needed in a very high frequency
+  if (entry_count++ % update_mem_usage_metrics_factor_ == 0) {
+    // enter here only once every update_mem_usage_metrics_factor_ calls of updateMetrics()
+    updateDBMemUsageMetrics();
+  }
+
+  // update all tickers
+  for (auto &pair : active_tickers_) {
+    pair.second.Get().Set(statistics->getTickerCount(pair.first));
+  }
+  // upodate total size
+  total_db_disk_size_.Get().Set(sstFm->GetTotalSize());
+
+  metrics_.UpdateAggregator();
+}
+
+// updates the rocksDB mem usage metrics
+void Client::RocksDbStorageMetrics::updateDBMemUsageMetrics() {
+  // update indexes_and_filters_ram_usage
+  std::string indexes_and_filters_ram_usage_str;
+  owning_client_.dbInstance_->GetProperty("rocksdb.estimate-table-readers-mem", &indexes_and_filters_ram_usage_str);
+  size_t indexes_and_filters_ram_usage = std::stoull(indexes_and_filters_ram_usage_str);
+  indexes_and_filters_ram_usage_.Get().Set(indexes_and_filters_ram_usage);
+
+  // update memtables_ram_usage
+  std::string memtables_ram_usage_str;
+  owning_client_.dbInstance_->GetProperty("rocksdb.cur-size-all-mem-tables", &memtables_ram_usage_str);
```

Comment:
I see that later down the code you dive into CF resolution.
how about adding 2 other properties:
```
"rocksdb.block-cache-usage"
"rocksdb.block-cache-pinned-usage",
```
If you think that bellow code already contains the same properties, just in mor detailed way, please ignore this comment.

In addition, you can go into RocksDB tag       v6.8.1 and look at:
rocksdb/include/rocksdb/db.h,  struct Properties

There are many properties which you can add and print with GetProperty, if you want to have more relevant information. I didn't look further into it.